src/main/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceAdapter.java

```diff
@@ -330,7 +333,17 @@ public VAppTemplate getImage(final String imageId) {
 
    @Override
    public Iterable<Vm> listNodes() {
-      return Sets.newHashSet();
+      Set<Vm> vms = Sets.newHashSet();
+      QueryResultRecords result = api.getQueryApi().vmsQueryAll();
+      for (QueryResultRecordType record : result.getRecords()) {
+         if (record instanceof QueryResultVMRecord) {
+            QueryResultVMRecord queryResultVMRecord = (QueryResultVMRecord) record;
+            if (!queryResultVMRecord.isVAppTemplate()) { // && queryResultVMRecord.getStatus().equals("POWERED_ON")) {
```

Comment:
Why the commented out code? I presume we want all nodes, not just the powered on ones so delete that?